sklearn/utils/_metadata_requests.py

```diff
@@ -0,0 +1,1266 @@
+    # corresponding method's signature.
+    UNUSED = "$UNUSED$"
+    # this is used whenever a default value is changed, and therefore the user
+    # should explicitly set the value, otherwise a warning is shown. An example
+    # is when a meta-estimator is only a router, but then becomes also a
+    # consumer in a new release.
+    WARN = "$WARN$"
+
+    @classmethod
+    def is_alias(cls, item):
+        """Check if an item is a valid alias.
+
+        Parameters
+        ----------
+        item : object
+            The given item to be checked if it can be an alias.
+
+        Returns
+        -------
+        result : bool
+            Whether the given item is a valid alias.
+        """
+        try:
+            cls(item)
+        except ValueError:
+            # item is only an alias if it's a valid identifier
+            return isinstance(item, str) and item.isidentifier()
+        else:
+            return False
+
+    @classmethod
+    def is_valid(cls, item):
+        """Check if an item is a valid RequestType (and not an alias).
+
+        Parameters
+        ----------
+        item : object
+            The given item to be checked.
+
+        Returns
+        -------
+        result : bool
+            Whether the given item is valid.
+        """
+        try:
+            cls(item)
+            return True
+        except ValueError:
+            return False
+
+
+# this is the default used in `set_{method}_request` methods to indicate no change
+# requested by the user.
+UNCHANGED = "$UNCHANGED$"
+
+# Only the following methods are supported in the routing mechanism. Adding new
+# methods at the moment involves monkeypatching this list.
+METHODS = [
+    "fit",
+    "partial_fit",
+    "predict",
+    "predict_proba",
+    "predict_log_proba",
+    "decision_function",
+    "score",
+    "split",
+    "transform",
+    "inverse_transform",
+]
+
+
+# These strings are used to dynamically generate the docstrings for
+# set_{method}_request methods.
+REQUESTER_DOC = """        Request metadata passed to the ``{method}`` method.
+
+        Note that this method is only relevant if
+        ``enable_metadata_routing=True`` (see :func:`sklearn.set_config`).
+        Please see :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        The options for each parameter are:
+
+        - ``True``: metadata is requested, and \
+passed to ``{method}`` if provided. The request is ignored if \
+metadata is not provided.
+
+        - ``False``: metadata is not requested and the meta-estimator \
+will not pass it to ``{method}``.
+
+        - ``None``: metadata is not requested, and the meta-estimator \
+will raise an error if the user provides it.
+
+        - ``str``: metadata should be passed to the meta-estimator with \
+this given alias instead of the original name.
+
+        The default (``UNCHANGED``) retains the existing request. This allows
```

Comment:
Saying that is is `UNCHANGED` when the actual default is `$UNCHANGED$` can be confusing.

BTW, I wish we had another sentinel that is meaningless in this context. Maybe the integer -1 ? (I do not really like it too much either).

Reply:
changed it to `sklearn.utils.metadata_routing.UNCHANGED`.